node/syncblock.go

```diff
@@ -192,6 +196,10 @@ func (localNode *LocalNode) getBlocksMessageHandler(remoteMessage *RemoteMessage
 		return replyBuf, false, nil
 	}
 
+	if !localNode.syncBlockLimiter.AllowN(time.Now(), int(endHeight-startHeight)) {
```

Comment:
If node received x block(x<endHeight-startHeight), it will return false, is it correct?

Reply:
Actually not, this is a rate limiter: https://pkg.go.dev/golang.org/x/time/rate?tab=doc It only allows other nodes to request N blocks per second on average, optionally with a burst